examples/applications/plot_cyclical_feature_engineering.py

```diff
@@ -0,0 +1,765 @@
+    RidgeCV(alphas=alphas),
+)
+evaluate(cyclic_spline_poly_pipeline, X, y, cv=ts_cv)
+
+# %%
+#
+# We observe that this model can almost rival the performance of the gradient
+# boosted trees with an average error around 6% of the maximum demand.
+#
+# Note that while the final step of this pipeline is a linear regression model,
+# the intermediate steps such as the spline feature extraction and the Nyström
+# kernel approximation are highly non-linear. As a result the compound pipeline
+# is much more expressive than a simple linear regression model with raw features.
+#
+# For the sake of completeness, we also evaluate the combination of one-hot
+# encoding and kernel approximation:
+
+one_hot_poly_pipeline = make_pipeline(
+    ColumnTransformer(
+        transformers=[
+            ("categorical", one_hot_encoder, categorical_columns),
+            ("one_hot_time", one_hot_encoder, ["hour", "weekday", "month"]),
+        ],
+        remainder="passthrough",
+    ),
+    Nystroem(kernel="poly", degree=2, n_components=300, random_state=0),
+    RidgeCV(alphas=alphas),
+)
+evaluate(one_hot_poly_pipeline, X, y, cv=ts_cv)
+
+
+# %%
+# While one-hot features were competitive with spline-based features when using
+# linear models, this is no longer the case when using a low-rank approximation
+# of a non-linear kernel: this can be explained by the fact that spline
+# features are smoother and allow the kernel approximation to find a more
+# expressive decision function.
+#
+# Let us now have a qualitative look at the predictions of the kernel models
+# and of the gradient boosted trees that should be able to better model
+# non-linear interactions between features:
+gbrt_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+gbrt_predictions = gbrt_pipeline.predict(X.iloc[test_0])
+
+one_hot_poly_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+one_hot_poly_predictions = one_hot_poly_pipeline.predict(X.iloc[test_0])
+
+cyclic_spline_poly_pipeline.fit(X.iloc[train_0], y.iloc[train_0])
+cyclic_spline_poly_predictions = cyclic_spline_poly_pipeline.predict(X.iloc[test_0])
+
+# %%
+# Again we zoom on the last 4 days of the test set:
+
+last_hours = slice(-96, None)
+fig, ax = plt.subplots(figsize=(12, 4))
+fig.suptitle("Predictions by non-linear regression models")
+ax.plot(
+    y.iloc[test_0].values[last_hours],
+    "x-",
+    alpha=0.2,
+    label="Actual demand",
+    color="black",
+)
+ax.plot(
+    gbrt_predictions[last_hours],
+    "x-",
+    label="Gradient Boosted Trees",
+)
+ax.plot(
+    one_hot_poly_predictions[last_hours],
+    "x-",
+    label="One-hot + polynomial kernel",
+)
+ax.plot(
+    cyclic_spline_poly_predictions[last_hours],
+    "x-",
+    label="Splines + polynomial kernel",
+)
+_ = ax.legend()
+
+
+# %%
+# First note that trees can naturally model non-linear feature interactions
+# since by default decision trees are allowed to grow beyond a depth of 2
+# levels.
```

Comment:
```suggestion
# First, note that trees can naturally model non-linear feature interactions
# since, by default, decision trees are allowed to grow beyond a depth of 2
# levels.
```